src/Microsoft.AspNet.Razor.Runtime/Resources.resx

```diff
@@ -156,4 +156,10 @@
   <data name="TagHelperDescriptorFactory_InvalidBoundAttributeName" xml:space="preserve">
     <value>Invalid tag helper bound property '{0}.{1}'. Tag helpers cannot bind to HTML attributes beginning with '{2}'.</value>
   </data>
+  <data name="TagHelperAttributes_CannotAddWithNullName" xml:space="preserve">
+    <value>Cannot add a '{0}' with a null '{1}'.</value>
+  </data>
+  <data name="TagHelperAttributes_CannotAddAttribute" xml:space="preserve">
+    <value>Cannot add an attribute with the name '{0}' when its name is '{1}'. The '{2}' property of the tag attribute must match the name it is added with.</value>
```

Comment:
first sentence in particular doesn't scan. suggest `Cannot add a {0} with inconsistent names. The {1} property '{2}' must match the location '{3}'.` (note quoting system-provided names isn't that useful.) details:
- 0 = `nameof(TagHelperAttribute)`
- 1 = `nameof(TagHelperAttribute.Name)`
- 2 = `attribute.Name`
- 3 = `index`